From fe9271ad84faa9f4739c855ec81abb9212527f74 Mon Sep 17 00:00:00 2001
From: Frantisek Sumsal <frantisek@sumsal.cz>
Date: Mon, 6 May 2019 10:48:12 +0200
Subject: [PATCH 26/29] test: return a non-zero return code when 'nobody' user
 doesn't exist

Lookup of a non-existing user using getpwnam() is not considered
an error, thus the `errno` is not set appropriately, causing
unexpected fails on systems, where 'nobody' user doesn't exist by
default

(cherry picked from commit 6ab668337e28ed8e594b0a9ce0b31f2afe3506d8)
---
 src/test/test-capability.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/test/test-capability.c b/src/test/test-capability.c
index 6b2de66bb7..f9fae84dde 100644
--- a/src/test/test-capability.c
+++ b/src/test/test-capability.c
@@ -99,7 +99,7 @@ static int setup_tests(bool *run_ambient) {
 
         nobody = getpwnam(NOBODY_USER_NAME);
         if (!nobody)
-                return log_error_errno(errno, "Could not find nobody user: %m");
+                return log_error_errno(SYNTHETIC_ERRNO(ENOENT), "Could not find nobody user: %m");
 
         test_uid = nobody->pw_uid;
         test_gid = nobody->pw_gid;
-- 
2.21.0

